pkg/runner/job_executor.go

```diff
@@ -55,42 +57,44 @@ func newJobExecutor(info jobInfo, sf stepFactory, rc *RunContext) common.Executo
 			return common.NewErrorExecutor(err)
 		}
 
-		preSteps = append(preSteps, step.pre())
+		preSteps = append(preSteps, useStepLogger(rc, stepModel, stepStagePre, step.pre()))
 
 		stepExec := step.main()
-		steps = append(steps, func(ctx context.Context) error {
-			stepName := stepModel.String()
-			return (func(ctx context.Context) error {
-				err := stepExec(ctx)
-				if err != nil {
-					common.Logger(ctx).Errorf("%v", err)
-					common.SetJobError(ctx, err)
-				} else if ctx.Err() != nil {
-					common.Logger(ctx).Errorf("%v", ctx.Err())
-					common.SetJobError(ctx, ctx.Err())
-				}
-				return nil
-			})(withStepLogger(ctx, stepName))
-		})
+		steps = append(steps, useStepLogger(rc, stepModel, stepStageMain, func(ctx context.Context) error {
+			logger := common.Logger(ctx)
+			err := stepExec(ctx)
+			if err != nil {
+				logger.Errorf("%v", err)
+				common.SetJobError(ctx, err)
+			} else if ctx.Err() != nil {
+				logger.Errorf("%v", ctx.Err())
+				common.SetJobError(ctx, ctx.Err())
+			}
+			return nil
+		}))
 
-		// run the post exector in reverse order
+		postExec := useStepLogger(rc, stepModel, stepStagePost, step.post())
 		if postExecutor != nil {
-			postExecutor = step.post().Finally(postExecutor)
+			// run the post exector in reverse order
+			postExecutor = postExec.Finally(postExecutor)
 		} else {
-			postExecutor = step.post()
+			postExecutor = postExec
 		}
 	}
 
 	postExecutor = postExecutor.Finally(func(ctx context.Context) error {
+		logger := common.Logger(ctx)
 		jobError := common.JobError(ctx)
 		if jobError != nil {
 			info.result("failure")
+			logger.WithField("jobResult", "failure").Infof("\U0001F3C1  Job failed")
 		} else {
 			err := info.stopContainer()(ctx)
 			if err != nil {
 				return err
 			}
 			info.result("success")
+			logger.WithField("jobResult", "success").Infof("\U0001F3C1  Job succeeded")
```

Comment:
I get a misleading logging message, this job doesn't succeed. I cancelled it.

```
🚀  Start image=ghcr.io/catthehacker/ubuntu:act-latest
  🐳  docker pull image=ghcr.io/catthehacker/ubuntu:act-latest platform= username= forcePull=false
🏁  Job succeeded
Warning: unable to get git repo: unable to find git repo
Error: context canceled
```
Probably a bug, which hasen't surfaced earlier.

I'm testing this change against my selfhosted runner code